File: composeApp/src/androidMain/kotlin/com/jetbrains/kmpapp/ui/theme/GyansarTheme.kt
package com.jetbrains.kmpapp.ui.theme

import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Typography
import androidx.compose.material3.darkColorScheme
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp

private val GyansarLightColorScheme = lightColorScheme(
    primary = Color(0xFF15803D),
    onPrimary = Color(0xFFFFFFFF),
    secondary = Color(0xFF84CC16),
    onSecondary = Color(0xFFFFFFFF),
    tertiary = Color(0xFF10B981),
    onTertiary = Color(0xFFFFFFFF),
    background = Color(0xFFFFFFFF),
    onBackground = Color(0xFF374151),
    surface = Color(0xFFF0FDF4),
    onSurface = Color(0xFF374151),
    surfaceVariant = Color(0xFFF0FDF4),
    onSurfaceVariant = Color(0xFF374151),
    outline = Color(0xFFE5E7EB),
    error = Color(0xFFFF4D4D),
    onError = Color(0xFFFFFFFF)
)

private val GyansarDarkColorScheme = darkColorScheme(
    primary = Color(0xFF22C55E),
    onPrimary = Color(0xFF0F172A),
    secondary = Color(0xFFA3E635),
    onSecondary = Color(0xFF0F172A),
    tertiary = Color(0xFF10B981),
    onTertiary = Color(0xFF0F172A),
    background = Color(0xFF0F172A),
    onBackground = Color(0xFFF8FAFC),
    surface = Color(0xFF1E293B),
    onSurface = Color(0xFFF8FAFC),
    surfaceVariant = Color(0xFF1E293B),
    onSurfaceVariant = Color(0xFF94A3B8),
    outline = Color(0xFF334155),
    error = Color(0xFFEF4444),
    onError = Color(0xFFF8FAFC)
)

private val GyansarTypography = Typography(
    displayLarge = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.SemiBold,
        fontSize = 34.sp,
        lineHeight = 40.sp
    ),
    headlineMedium = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.SemiBold,
        fontSize = 24.sp,
        lineHeight = 30.sp
    ),
    titleLarge = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.Medium,
        fontSize = 20.sp,
        lineHeight = 26.sp
    ),
    titleMedium = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.Medium,
        fontSize = 18.sp,
        lineHeight = 24.sp
    ),
    bodyLarge = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.Normal,
        fontSize = 16.sp,
        lineHeight = 22.sp
    ),
    bodyMedium = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.Normal,
        fontSize = 14.sp,
        lineHeight = 20.sp
    ),
    labelLarge = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.SemiBold,
        fontSize = 14.sp,
        lineHeight = 18.sp
    ),
    labelMedium = TextStyle(
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.Medium,
        fontSize = 12.sp,
        lineHeight = 16.sp
    )
)

@Composable
fun GyansarTheme(
    darkTheme: Boolean = isSystemInDarkTheme(),
    content: @Composable () -> Unit
) {
    MaterialTheme(
        colorScheme = if (darkTheme) GyansarDarkColorScheme else GyansarLightColorScheme,
        typography = GyansarTypography,
        content = content
    )
}

File: iosApp/iosApp/Theme/GyansarTheme.swift
import SwiftUI

class GyansarTheme: ObservableObject {
    @Published var colorScheme: ColorScheme = .light
    
    var primary: Color { colorScheme == .dark ? GyansarColors.Dark.primary : GyansarColors.Light.primary }
    var onPrimary: Color { colorScheme == .dark ? GyansarColors.Dark.onPrimary : GyansarColors.Light.onPrimary }
    var secondary: Color { colorScheme == .dark ? GyansarColors.Dark.secondary : GyansarColors.Light.secondary }
    var onSecondary: Color { colorScheme == .dark ? GyansarColors.Dark.onSecondary : GyansarColors.Light.onSecondary }
    var tertiary: Color { colorScheme == .dark ? GyansarColors.Dark.tertiary : GyansarColors.Light.tertiary }
    var onTertiary: Color { colorScheme == .dark ? GyansarColors.Dark.onTertiary : GyansarColors.Light.onTertiary }
    var background: Color { colorScheme == .dark ? GyansarColors.Dark.background : GyansarColors.Light.background }
    var onBackground: Color { colorScheme == .dark ? GyansarColors.Dark.onBackground : GyansarColors.Light.onBackground }
    var surface: Color { colorScheme == .dark ? GyansarColors.Dark.surface : GyansarColors.Light.surface }
    var onSurface: Color { colorScheme == .dark ? GyansarColors.Dark.onSurface : GyansarColors.Light.onSurface }
    var surfaceVariant: Color { colorScheme == .dark ? GyansarColors.Dark.surfaceVariant : GyansarColors.Light.surfaceVariant }
    var onSurfaceVariant: Color { colorScheme == .dark ? GyansarColors.Dark.onSurfaceVariant : GyansarColors.Light.onSurfaceVariant }
    var outline: Color { colorScheme == .dark ? GyansarColors.Dark.outline : GyansarColors.Light.outline }
    var error: Color { colorScheme == .dark ? GyansarColors.Dark.error : GyansarColors.Light.error }
    var onError: Color { colorScheme == .dark ? GyansarColors.Dark.onError : GyansarColors.Light.onError }
    
    func heroBrush() -> LinearGradient {
        LinearGradient(
            colors: [background, surface],
            startPoint: .top,
            endPoint: .bottom
        )
    }
}

struct GyansarThemeKey: EnvironmentKey {
    static let defaultValue = GyansarTheme()
}

extension EnvironmentValues {
    var gyansarTheme: GyansarTheme {
        get { self[GyansarThemeKey.self] }
        set { self[GyansarThemeKey.self] = newValue }
    }
}

struct ThemedView<Content: View>: View {
    @Environment(\.colorScheme) private var systemColorScheme
    @StateObject private var theme = GyansarTheme()
    let content: (GyansarTheme) -> Content
    
    var body: some View {
        content(theme)
            .environment(\.gyansarTheme, theme)
            .onAppear {
                theme.colorScheme = systemColorScheme
            }
            .onChange(of: systemColorScheme) { newScheme in
                theme.colorScheme = newScheme
            }
    }
}
